### PR TITLE
[webview_flutter_web] Avoids XHR when possible.

### PR DESCRIPTION
**Pull request by bparrishMines**
This PR:

* Updates `WebWebViewController.loadRequest` to only set the src of the iFrame when `LoadRequestParams.headers` and `LoadRequestParams.body` are empty and is using the HTTP GET request method.
* Parses the `content-type` header of XHR responses to extract the correct MIME-type and charset.
* Sets `width` and `height` of widget the way the Flutter engine wants, to remove distracting warnings from the development console.

Fixes https://github.com/flutter/flutter/issues/118573
Fixes https://github.com/flutter/flutter/issues/118090

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/main/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Review by ditman — APPROVED** [review]
This seems to work in my machine.

File: packages/webview_flutter/webview_flutter_web/example/integration_test/webview_flutter_test.dart [hunk 1]
```diff
@@ -5,6 +5,10 @@
 import 'dart:html' as html;
 import 'dart:io';
 
+// FIX (dit): Remove these integration tests, or make them run. They currently never fail.
```
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter_web/example/integration_test/webview_flutter_test.dart:8]: Ah, I didn't recognize this when I updated these tests. Thanks for noticing this!

**Review by bparrishMines — COMMENTED** [review]
LGTM as joint PR

**Comment by ditman** [comment]
Thanks for landing this @bparrishMines!

**Comment by bparrishMines** [comment]
Thanks for your help with the PR! I definitely didn't have the knowledge to fix these bugs without your help!

**Pull request merged**